crates/re_selection_panel/src/visualizer_ui.rs

```diff
@@ -154,22 +185,17 @@ fn visualizer_components(
 
         // Determine where the final value comes from.
         // Putting this into an enum makes it easier to reason about the next steps.
-        let value_source = match (non_empty_override, non_empty_store, non_empty_default) {
-            (true, _, _) => ValueSource::Override,
-            (false, true, _) => ValueSource::Store,
-            (false, false, true) => ValueSource::Default,
-            (false, false, false) => ValueSource::FallbackOrPlaceholder,
+        let (value_source, raw_current_value) = match (
+            raw_override.as_ref(),
+            raw_store.as_ref(),
+            raw_default.as_ref(),
+        ) {
+            (Some(override_value), _, _) => (ValueSource::Override, override_value.as_ref()),
+            (None, Some(store_value), _) => (ValueSource::Store, store_value.as_ref()),
+            (None, None, Some(default_value)) => (ValueSource::Default, default_value.as_ref()),
+            (None, None, None) => (ValueSource::FallbackOrPlaceholder, raw_fallback.as_ref()),
         };
```

Comment:
For my own comprehension: this bit duplicates the logic of how things happen for real, but can't be DRYed because the actual logic is dispatched all over the place (with fallback providers, etc.). Correct?

Reply:
yes, correct.
The "for real" logic itself is unfortunately a bit cluttered and needs cleanup. Luckily we arrived at this relatively simple stack. It would be still good to be able to unify this more, especially given how annotation context doesn't fit into this at all right now :/